After enable R8 full mode getting ParameterizedType error

**Please complete the following information:**
- Library Version: 1.3.5
 
**Describe the Bug:**

This issue is similar to the issue in Retrofit (https://github.com/square/retrofit/issues/3751) but I'm not sure, but I've duplicated it here. This happens with AGP 8.0. As a solution, it's proposed to add these proguard lines to the project (https://github.com/square/retrofit/issues/3751#issuecomment-1192043644):
```
 # Keep generic signature of Call, Response (R8 full mode strips signatures from non-kept items). 
 -keep,allowobfuscation,allowshrinking interface retrofit2.Call 
 -keep,allowobfuscation,allowshrinking class retrofit2.Response 
  
 # With R8 full mode generic signatures are stripped for classes that are not 
 # kept. Suspend functions are wrapped in continuations where the type argument 
 # is used. 
 -keep,allowobfuscation,allowshrinking class kotlin.coroutines.Continuation 
```

but it doesn't actually work, After adding this proguard lines I get a slightly different error:
```
 java.lang.IllegalArgumentException: Unable to create call adapter for retrofit2.Call<com.skydoves.sandwich.ApiResponse>
                     for method ComicVineService.issues
                 	at retrofit2.Utils.methodError(SourceFile:47)
                 	at retrofit2.HttpServiceMethod.parseAnnotations(SourceFile:394)
                 	at retrofit2.Retrofit.loadServiceMethod(SourceFile:31)
                 	at retrofit2.Retrofit$1.invoke(SourceFile:45)
                 	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
                 	at $Proxy9.issues(Unknown Source)
                        ...
Caused by: java.lang.ClassCastException: java.lang.Class cannot be cast to java.lang.reflect.ParameterizedType
                 	at com.skydoves.sandwich.adapters.ApiResponseCallAdapterFactory.get(SourceFile:53)
                 	at retrofit2.Retrofit.callAdapter(SourceFile:33)
                 	at retrofit2.HttpServiceMethod.parseAnnotations(SourceFile:375)
                 	... 41 more
```

```Kotlin
 @GET("issues?format=$FORMAT&field_list=${Fields.Issues}")
    @JvmSuppressWildcards
    suspend fun issues(
        @Query("api_key") apiKey: String,
        @Query("offset") offset: Int,
        @Query("limit") limit: Int,
        @Query("sort", encoded = true) sort: ComicVineSort?,
        @Query("filter[]", encoded = true) filter: List<ComicVineFilter>?,
    ): ApiResponse<IssuesResponse>
```

--- a/README.md
+++ b/README.md
@@ -56,6 +56,11 @@ dependencies {
 
 </details>
 
+
+## R8 / ProGuard
+The specific rules are [already bundled](sandwich/src/main/resources/META-INF/proguard/sandwich.pro) into the JAR which can be interpreted by R8 automatically.
+
+
 ## Use Cases
 You can also check out nice use cases of this library in the repositories below:
 - [Pokedex](https://github.com/skydoves/pokedex): 🗡️ Android Pokedex using Hilt, Motion, Coroutines, Flow, Jetpack (Room, ViewModel, LiveData) based on MVVM architecture.

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -39,6 +39,14 @@ android {
     dataBinding true
   }
 
+  buildTypes {
+    release {
+      minifyEnabled true
+      proguardFiles getDefaultProguardFile('proguard-android.txt')
+      signingConfig = signingConfigs.debug
+    }
+  }
+
   // Can remove this once https://issuetracker.google.com/issues/260059413 is fixed.
   compileOptions {
     sourceCompatibility JavaVersion.VERSION_11

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -42,7 +42,7 @@ android {
   buildTypes {
     release {
       minifyEnabled true
-      proguardFiles getDefaultProguardFile('proguard-android.txt')
+      proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
       signingConfig = signingConfigs.debug
     }
   }

--- a/app/proguard-rules.pro
+++ b/app/proguard-rules.pro
@@ -1,0 +1,10 @@
+# https://github.com/square/okhttp/blob/339732e3a1b78be5d792860109047f68a011b5eb/okhttp/src/jvmMain/resources/META-INF/proguard/okhttp3.pro#L11-L14
+-dontwarn okhttp3.internal.platform.**
+-dontwarn org.conscrypt.**
+-dontwarn org.bouncycastle.**
+-dontwarn org.openjsse.**
+
+# TODO: Waiting for new retrofit release to remove these rules
+-keep,allowobfuscation,allowshrinking interface retrofit2.Call
+-keep,allowobfuscation,allowshrinking class retrofit2.Response
+-keep,allowobfuscation,allowshrinking class kotlin.coroutines.Continuation

--- a/sandwich/src/main/resources/META-INF/proguard/sandwich.pro
+++ b/sandwich/src/main/resources/META-INF/proguard/sandwich.pro
@@ -1,0 +1,2 @@
+# R8 full mode strips signatures from non-kept items.
+-keep,allowobfuscation,allowshrinking class com.skydoves.sandwich.ApiResponse

